chapters/06_agent_based_extension/01_state_dependent/main.tex

```diff
@@ -0,0 +1,148 @@
+\section{State-dependent variation}\label{sec:state_dependent}
+
+This section focuses on creating a state-dependent variation of the queueing
+system described in Section~\ref{sec:queueing_section}.
+The state-dependent version is based on the idea that the speed of the servers
+could be based on the number of individuals present in the system.
+This was implemented only on the Discrete Event Simulation (DES) version of
+the queueing model described in Section~\ref{sec:discrete_event_simulation}.
+
+The new service rate \(\mu\) that is used here is defined as follows:
+
+\begin{equation}\label{eq:state_dependent_service_rate}
+    \mu =
+    \begin{cases}
+        \mu_{(0,0)}, & \text{if } (u, v) = (0, 0) \\
+        \mu_{(0,1)}, & \text{if } (u, v) = (0, 1) \\
+        \quad \vdots & \qquad \quad \vdots \\
+        \mu_{(N,M)}, & \text{if } (u, v) = (M, N)
+    \end{cases}
+\end{equation}
+
+Although the state-dependent variation of the queueing system was implemented
+only on the DES version of the queue, for visualisation purposes, the Markov
+chain model will be used to better demonstrate the concept of this model.
+Consider an example of the queueing system described in
+Section~\ref{sec:queueing_section} where the number of servers is set to
+\(C = 1\), the threshold is set to \(T = 1\), node 1 capacity is \(N = 2\) and
+node 2 capacity is \(M = 1\).
+Figure~\ref{fig:state_dependent_markov_example} shows the state-dependent
+Markov chain model of the queueing system.
+
+\begin{figure}[H]
+    \centering
+    \input{chapters/06_agent_based_extension/Bin/example_model_1121/main.tex}
+    \caption{Markov chain example with \(C=1, T=1, N=2, M=1\)}
+    \label{fig:state_dependent_markov_example}
+\end{figure}
+
+In the previous chapters of this thesis, the value of the service rate \(\mu\)
+was set to a constant value.
+For this variation of the model \(\mu\) will have a different value depending
+on the number of individuals present in the two nodes of the queue.
+In other words \(\mu\) will take a different value depending on the state
+\(u, v\) of which the system is in.
```

Comment:
Move this to before the definition of \(\mu\)

Reply:
Will do 👍